Add humidity compensation

mguntli opened the pull request:
The SGP30 features an on-chip humidity compensation for the air quality signals (CO2eq and TVOC) and sensor raw signals (H2-signal and Ethanol_signal).
To use the on-chip humidity compensation an absolute humidity value from an external humidity sensor like the SHTxx is required.

mguntli committed:
Add humidity compensation

The SGP30 features an on-chip humidity compensation for the air quality signals (CO2eq and TVOC) and sensor raw signals (H2-signal and Ethanol_signal).
To use the on-chip humidity compensation an absolute humidity value from an external humidity sensor like the SHTxx is required.

mdeyo commented:
@mguntli What's your progress on addressing the CI errors? I was about to implement the 'set_Humidity' function as documented in the SGP30 datasheet and could help with this request.

mguntli commented:
My implementation is according to the SGP30 datasheet (see function _setHumidity_).
I have forked an own SGP30 library for the particle platform (https://github.com/mguntli/particle-lib-sgp30), and **wanted to share back my additions with this pull request**.

I would be happy if you could have a look at the CI errors, thanks for that!

Note: The approximation formula to get the absolute humidity [mg/m^3] based on temperature [°C] and humidity [%RH] is the recommended one from Sensirion SGP30 Driver Integration chapter 3.16
https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/9_Gas_Sensors/Sensirion_Gas_Sensors_SGP30_Driver-Integration-Guide_SW_I2C.pdf


![equation](http://latex.codecogs.com/gif.latex?AH%20%3D%20216.7%20*%20%5Cfrac%7B%5Cfrac%7BRH%7D%7B100.0%7D%20*%206.112%20*%20exp%5Cfrac%7B17.62*t%7D%7B243.12&plus;t%7D%7D%7B273.15&plus;t%7D)  
with AH in g/m^3, RH in 0-100%, and t in °C

caternuson commented on Adafruit_SGP30.cpp:
```diff
@@ -156,6 +156,29 @@ boolean Adafruit_SGP30::setIAQBaseline(uint16_t eco2_base, uint16_t tvoc_base) {
+    @returns True if command completed successfully, false if something went wrong!
+*/
+/**************************************************************************/
+boolean SensirionSGP30::setHumidity(uint32_t absolute_humidity) {
```
Need to change class name to Adafruit_SGP30

mdeyo replied:
This should be the only real error. @mguntli would it take much effort to fix the class name and try again?

mguntli replied:
Thanks for the hint!
The errors have been introduced to renaming on my own fork for the particle library..

ladyada commented:
hiya just a note - if this isnt compiling, it wasnt tested & we wouldnt merge unless it was tested!

mdeyo commented:
Ordered the breakout board because of the SGP30 setHumidity feature and was disappointed to not see it exposed yet. Branch author has setHumidity interface working on his own forked project but renamed the class. Just trying to get that function where more people can use it (also addresses #3).

mguntli committed:
Fix compile error due to wrong class name

Replaced SensirionSGP30 with Adafruit_SGP30 (errors introduced to renaming on my own fork for the particle library)

mguntli commented:
I can confirm that I have tested the humidity compensation feature on my Adafruit SGP30, but only with my own fork of the library with different class naming (https://github.com/mguntli/particle-lib-sgp30/pull/1).

SGP30 and SHT31-D:
![SGP30 and SHT31-D](https://i.imgur.com/kRtSoge.jpg)

Absolute Humidity Values in mg/m^3:
![Absolute Humidity Values](https://i.imgur.com/bO3r857.png)

All Values of SGP30 and STH31-D:
![All Values](https://i.imgur.com/cIpQTIm.png)

mguntli commented:
@ladyada Any chance you will merge this pull request?

ladyada commented:
it would need to be tested with a sensor and arduino...if you do that on this repo please let us know!

mdeyo commented:
@ladyada I finally had time to test the code from his branch "mguntli:feature/adafruit-humidity-compensation" (using commit d5e793e5e0500feb442de5760b8b12b36b49848d) on a Huzzah ESP8266 (using Arduino IDE). It works without errors, both with the setHumidity function commented out (default in his sgp30test example) and with it using the given constant temp+humidity sample values. I also printed the boolean responses from readWordFromCommand to ensure the commands in setHumidity were working. Let me know if you want additional proof of my testing/verification! (Like do you also want it tested on an Arduino?)

ladyada commented:
sounds good! pulled :) thank you!!!
